### ISSUE TITLE
Sparse Tensors

### ISSUE THREAD

**Issue by kamathis4** [keras-team/keras#18420]
Hey @fchollet,

do you have any roadmap to implement sparse tensors for the different backends?
- tf seems to have only one implementation of sparse tensors [COO](https://www.tensorflow.org/guide/sparse_tensor)
- torch seems to have multiple methods but has really good support for [COO](https://pytorch.org/docs/stable/sparse.html#torch-functions-specific-to-sparse-tensors)
- I think for numpy mostly through [scipy](https://docs.scipy.org/doc/scipy/reference/sparse.html)
- JAX has sparse support seems [experimental](https://jax.readthedocs.io/en/latest/jax.experimental.sparse.html) for now

**Comment by ghsanti** [comment]
Hi!

I wonder whether there are any updates to support sparse tensors for back-ends other than Tensorflow @fchollet ?

**Comment by hertschuh** [comment]
> Hi!
> 
> Are there any updates for other-than-tf support for sparse tensors @fchollet ?

@ghsanti ,

Sparse support has been added for JAX using `jax.experimental.sparse.BCOO` specifically. Feature-wise, it's at parity with TensorFlow for most things (gradients is the most notable gap).

**Comment by ghsanti** [comment]
> > Hi!
> > 
> > Are there any updates for other-than-tf support for sparse tensors @fchollet ?
> 
> @ghsanti ,
> 
> Sparse support has been added for JAX using `jax.experimental.sparse.BCOO` specifically. Feature-wise, it's at parity with TensorFlow for most things (gradients is the most notable gap).

I need the sparse gradients (not dense) as that would otherwise defeat its purpose? (Memory would go up, speed down.)

With this constraint on gradients, is tf the only backend possible? 

Though assuming "yes", although neither sparse graph sparse convolutions nor convolutions are available layers.

Graph-like network seems implementable in a layer using smth like $\sigma (AW)$ since keras supports sparse matmul through tf, right? And get the sparse gradient if I get your point correctly?

**Comment by hertschuh** [comment]
@ghsanti 

JAX handles sparseness in gradients differently, so there's nothing we can do in Keras right now. That makes TF the only possible backend. Sparse convolutions are indeed not available in Tensorflow. Sparse matmul does produce sparse gradients, so you could try that.

**Comment by dhantule** [comment]
@adi-kmt, Thanks for reporting.  Could you please close the issue, if you don't have anymore questions ? Thanks

**Comment by hertschuh** [comment]
Sparse support was added for Tensorflow and JAX.

However, as mentioned above, only Tensorflow supports sparse gradients.
